cmd/wallet/main.go

```diff
@@ -1,163 +1,391 @@
+	"github.com/tyler-smith/go-bip39"
+	"github.com/wavesplatform/gowaves/pkg/crypto"
+	"github.com/wavesplatform/gowaves/pkg/proto"
 	"github.com/wavesplatform/gowaves/pkg/wallet"
 )
 
-var usage = `
+const (
+	newOpt               = "new"
+	showOpt              = "show"
+	seedPhraseOpt        = "seed-phrase"
+	seedPhraseBase58Opt  = "seed-phrase-base58"
+	accountSeedBase58Opt = "account-seed-base58"
+
+	schemeOpt = "scheme"
+)
 
+const (
+	defaultBitSize    = 160
+	walletDefaultName = ".waves"
+)
+
+var usage = `
 Usage:
-  wallet command [flags]
+  wallet [flags]
 
-Available Commands:
-  add          Add seed to wallet
-  show         Print wallet data
+Flags:
+`
 
+var examples = `
+Examples:
+	./wallet -show -wallet <wallet path> // Show existing wallet credentials
+	./wallet -new // Generate a seed phrase and a wallet 
+	./wallet -seed-phrase "..." // Import a seed phrase
+	./wallet -seed-phrase-base58 "..." // Import a Base58 encoded seed phrase
+	./wallet -account-seed "..." // Import a Base58 encoded account seed
 `
 
 type Opts struct {
-	Force        bool
-	PathToWallet string
-	Base58       bool
+	PathToWallet       string
+	AccountNumber      int
+	Scheme             proto.Scheme
+	SeedPhrase         string
+	IsSeedPhraseBase58 bool
+	Base58AccountSeed  string
+	New                bool
+	Show               bool
+}
+
+type SchemeFlag byte
+
+func (sch *SchemeFlag) String() string {
+	str := string(*sch)
+	return str
+}
+func (sch *SchemeFlag) Set(s string) error {
+	val := []byte(s)
+	if len(val) > 1 {
+		return errors.New("failed to parse scheme: more than one letter was provided")
+	}
+
+	*sch = SchemeFlag(val[0])
+	return nil
 }
 
 func main() {
 	opts := Opts{}
-
-	flag.BoolVarP(&opts.Force, "force", "f", false, "Overwrite existing wallet")
-	flag.StringVarP(&opts.PathToWallet, "wallet", "w", "", "Path to wallet")
-	flag.BoolVarP(&opts.Base58, "base58", "b", false, "Input seed as Base58 encoded string")
+	var scheme SchemeFlag
+	flag.BoolVar(&opts.New, newOpt, false, "Generate and add a new seed phrase (Primary flag)")
+	flag.BoolVar(&opts.New, showOpt, false, "Show existing wallet credentials (Primary flag)")
+	flag.StringVar(&opts.SeedPhrase, seedPhraseOpt, "", "Import a seed phrase (Primary flag)")
+	flag.BoolVar(&opts.IsSeedPhraseBase58, seedPhraseBase58Opt, false, "Import a base58-encoded seed phrase (Primary flag)")
+	flag.StringVar(&opts.Base58AccountSeed, accountSeedBase58Opt, "", "Import a base58-encoded account seed (Primary flag)")
+	flag.StringVar(&opts.PathToWallet, "wallet", "", "Path to the wallet file")
+	flag.IntVar(&opts.AccountNumber, "number", 0, "Account number. 0 is default")
+	flag.Var(&scheme, schemeOpt, "Network scheme: MainNet=W, TestNet=T, StageNet=S, CustomNet=E. MainNet is default")
 
 	flag.Parse()
 
-	command := flag.Arg(0)
+	possibleFlags := []string{newOpt, showOpt, seedPhraseOpt, seedPhraseBase58Opt, accountSeedBase58Opt}
+	if moreThanOnePrimaryFlagPassed(possibleFlags) {
```

Comment:
All this flags to commands logic is overcomplicated. I think `moreThanOnePrimaryFlagPassed` check is enough.

Reply:
I wanted to have only one point of creating the possible flags, so if we ever add / remove some of them, we wouldn't have to change it in every function

Reply:
Moved it to global var array